device/simulation/tt_simulation_host.hpp

```diff
@@ -4,13 +4,13 @@
 #include <vector>
 #include <cstdint>
```

Comment:
```suggestion
#include <memory>
```

include-what-you-use for safety :-)